### ISSUE TITLE
General instability since version 6.2.6

### ISSUE THREAD

**Issue by DClabaut** [RocketChat/Rocket.Chat#30850]
### Description:

I am running a RocketChat server since early 2019. It's been working fine until I upgraded it from 5.1.4 to 6.2.6 back in June 2023. I have since updated to 6.4.2, but the instability persists.

Symptoms include:
- After pressing Enter to send a message, the message does not appear in the chat (channels, threads DM alike). It's not possible to edit it with "arrow up". It's just lost.
- Opening a channel doesn't mark it as read. Sometime forcing "mark as read" from the sidebar does not work.
- Uploading a picture  or file fails silently, it does not show the interface to change the description before posting.
- Channels / threads stop updating.

All of these happen without explicit errors, and the only solution is to fully reload the page.

I have tried the following improvements:
- Scaling the VM, which now has 4 cores and 24 GiB RAM
- Replace Apache with Nginx as reverse-proxy
- Running 3 instances of RocketChat (Systemd services) following [this doc](https://docs.rocket.chat/deploy/deploy-rocket.chat/scaling-rocket.chat/running-multiple-instances)
- Moving to faster storage (the VM runs off a NVMe SSD local to the hypervisor)

And the following troubleshooting:
- Enable Prometheus to export the app metrics to our Grafana instance.
- Check logs to look for errors.
- Check monitoring from the VM / network itself.
- Check the browser console to look for errors.

The issue is still there.

It's a rather small instance. We have about 25 active users and another 25 away during the day. The whole MongoDB weighs 60 GiB (there's years of stuff in there).

Load on the VM is minimal:
![Capture d’écran du 2023-11-03 11-32-16](https://github.com/RocketChat/Rocket.Chat/assets/965290/8b28aa16-3791-4f48-a153-c7eea33ad8b6)

Load on the app itself is minimal:
![Capture d’écran du 2023-11-03 11-15-10](https://github.com/RocketChat/Rocket.Chat/assets/965290/6ef6aabf-a790-4310-af7f-773935282905)

![Capture d’écran du 2023-11-03 11-28-04](https://github.com/RocketChat/Rocket.Chat/assets/965290/c1fceab0-c1af-4ce8-8aaa-762561bab22a)

App monitoring show it's responsive (Meteor Methods Total Time):
![Capture d’écran du 2023-11-03 11-29-18](https://github.com/RocketChat/Rocket.Chat/assets/965290/11a535dd-f4a7-4916-aed9-395bb5c74644)

The "Messages round trip" value seems of high, but I'm unsure about what this is exactly since this value is missing from [the doc](https://github.com/RocketChat/Rocket.Chat.Metrics/blob/master/metrics.md).
![Capture d’écran du 2023-11-03 11-28-43](https://github.com/RocketChat/Rocket.Chat/assets/965290/c35033d3-11de-4f85-b1ed-e83351efc06d)

### Steps to reproduce:

Issue appears randomly but often since the upgrade to 6.x.x.

### Expected behavior:

RocketChat is as stable as it was on version 5.x.

### Server Setup Information:

- Version of Rocket.Chat Server: 6.4.2
- Operating System:  Debian 11.8
- Deployment Method: Tar
- Number of Running Instances: 3 instances on same 4-core VM
- DB Replicaset Oplog: 
- NodeJS Version: 14.21.3
- MongoDB Version: 5.0.18 / wiredTiger (oplog Enabled)

### Client Setup Information

- Desktop App or Browser Version: Reproduced in recent version of Chrome and Firefox, as well as native apps (Linux, Android, IOS).
- Operating System:

### Additional context

The issue has been noticed by most users regardless of OS, browser, physical location...

**Comment by DClabaut** [comment]
On a copy of the VM, I moved the storage from gridFS to Filesystem using https://github.com/arminfelder/gridfsmigrate (I had to remove references to usernames and passwords to get it to run). Somehow removing 36 GiB of files from the DB allowed it to shrink from 60 GiB to 6 GiB. The issues are still there.

**Comment by fvlasie** [comment]
I am also experiencing similar issues with stability.

**Comment by RAurelian** [comment]
We are also experiencing major performance issues after updating our Snaps deployment to version 6.4.x. To put things into perspective, 3 months ago we had ~ 500 users very active on the same deployment without absolutely no performance issues on the 6.2.x version, but now even with ~ 250 users RC is almost unusable.

**Comment by DClabaut** [comment]
I disagree that it's a performance issue. I can reproduce at least some of the bugs (like being unable to edit a message, as the edited message won't send) while being the only connected user on the dev instance.

**Comment by ech0r** [comment]
Not sure what you expect, JS development and their tooling/dependencies are a house of cards. The docs on this project aren't event maintained.

**Comment by DClabaut** [comment]
As a follow-up, we deactivated Link Embeds (in Administration > Workspace > Settings > Messages) and this happened.

![Capture d’écran du 2024-01-30 14-58-37](https://github.com/RocketChat/Rocket.Chat/assets/965290/4fe76ee1-87b7-4e8f-9d8f-1adda73aaaf7)

![Capture d’écran du 2024-01-30 14-58-50](https://github.com/RocketChat/Rocket.Chat/assets/965290/38f2e278-d568-4abe-92e8-5d68f1c1a435)

The whole instance is noticeably faster now.
